packages/lib/schedules/updateSchedule.ts

```diff
@@ -65,7 +65,11 @@ export const updateSchedule = async ({ input, user, prisma }: IUpdateScheduleOpt
     updatedUser = setupDefault;
   }
 
-  if (!input.name) {
+  if (input.name !== undefined && input.name.trim().length === 0) {
+    throw new TRPCError({ code: "BAD_REQUEST", message: "Schedule name cannot be empty" });
+  }
+
+  if (input.name === undefined && !input.schedule && !input.timeZone) {
```

Comment:
```suggestion
  if (!input.name) {
```

I think we can revert it back to simply checking for `!input.name`, unless there is some reason you are also checking for `!input.schedule && !input.timeZone`?